src/backups.py

```diff
@@ -573,6 +605,40 @@ def _retrieve_s3_parameters(self) -> Tuple[Dict, List[str]]:
 
         return s3_parameters, []
 
+    def start_stop_pgbackrest_service(self) -> bool:
+        """Start or stop the pgBackRest TLS server service.
+
+        Returns:
+            a boolean indicating whether the operation succeeded.
+        """
+        # Ignore this operation if backups settings aren't ok.
+        are_backup_settings_ok, _ = self._are_backup_settings_ok()
+        if not are_backup_settings_ok:
+            return True
+
+        # Update pgBackRest configuration (to update the TLS settings).
+        if not self._render_pgbackrest_conf_file():
+            return False
+
+        snap_cache = snap.SnapCache()
+        charmed_postgresql_snap = snap_cache["charmed-postgresql"]
+        if not charmed_postgresql_snap.present:
+            logger.error("Cannot start/stop service, snap is not yet installed.")
+            return False
+
+        # Stop the service if TLS is not enabled or there are no replicas.
+        if not self.charm.is_tls_enabled or len(self.charm._peer_members_ips) == 0:
```

Comment:
do we need pgbackrest running if cluster size==1?
We should allow backups if it is 1 member cluster

Reply:
No, we don't need. The service is needed only when the user wants to backup from a replica (cluster size > 1).